### ISSUE TITLE
debian builds are failing on KeyError: "The cache has no package named 'apt-src'"

### ISSUE THREAD

**Issue by tfoote** [ros-infrastructure/ros_buildfarm#369]
KeyError: "The cache has no package named 'apt-src'"

This is happening across many jobs, some are flagged in #368 I've also seen completely unreleated jobs: http://build.ros.org/job/Kbin_dj_dJ64__cl_tf2__debian_jessie_amd64__binary/13/console

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apt/cache.py", line 201, in __getitem__
    return self._weakref[key]
  File "/usr/lib/python3.4/weakref.py", line 125, in __getitem__
    o = self.data[key]()
KeyError: 'apt-src'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/ros_buildfarm/scripts/release/create_binarydeb_task_generator.py", line 163, in <module>
    main()
  File "/tmp/ros_buildfarm/scripts/release/create_binarydeb_task_generator.py", line 85, in main
    apt_cache, debian_pkg_names)
  File "/tmp/ros_buildfarm/ros_buildfarm/common.py", line 134, in get_binary_package_versions
    pkg = apt_cache[debian_pkg_name]
  File "/usr/lib/python3/dist-packages/apt/cache.py", line 208, in __getitem__
    raise KeyError('The cache has no package named %r' % key)
KeyError: "The cache has no package named 'apt-src'"
```

**Comment by dirk-thomas** [comment]
If the `apt update` command uses `http://http.debian.net` the apt cache doesn't contain an entry for the `apt-src` package (and our script fails with the `KeyError`). If it uses `http://deb.debian.org` the apt cache has the expected entry and the build succeeds.

Could it be that we are randomly hitting an inconsistent Debian mirror?

**Comment by dirk-thomas** [comment]
#370 switches to use `deb.debian.org`. We will merge that and see in the next days if the problem happens again.

**Comment by tfoote** [comment]
Unfortunately it seems that there's still a random mirror failing. I just noticed another failure here and it only pulls from http://deb.debian.org : http://build.ros.org/view/Kbin_djv8_dJv8/job/Kbin_djv8_dJv8__agvs_description__debian_jessie_arm64__binary/8/console

**Comment by dirk-thomas** [comment]
We will merge this before the automatic reconfigure of the source jobs tonight.

**Comment by dirk-thomas** [comment]
Since I can't figure out why this is happening sporadically, can't reproduce it, and can't find similar reports online I created #387 as a workaround which should make sure that `apt-src` is in the cache.

**Comment by dirk-thomas** [comment]
Hopefully addressed by #387. Please comment if you see this failure again and include a link to the failing build.
